Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/FileTransporter.cs

```diff
@@ -5,461 +5,472 @@
-        {
-            var userResultMessage = string.Empty;
-
-            var errorMessages = results.SelectMany(x => x.ErrorMessages).ToList();
-            if (errorMessages.Any())
-                userResultMessage = MessageJoin(userResultMessage,
-                    string.Format("{0} Errors: {1}", errorMessages.Count, string.Join(", ", errorMessages)));
-
-            var transferredFiles = results.Select(x => x.TransferredFile).Where(x => x != null).ToList();
-            if (transferredFiles.Any())
-                userResultMessage = MessageJoin(userResultMessage,
-                    string.Format("{0} files transferred: {1}", transferredFiles.Count,
-                        string.Join(", ", transferredFiles)));
-            else
-                userResultMessage = MessageJoin(userResultMessage, "No files transferred.");
+    private FileTransferResult FormResultFromSingleTransferResults(List<SingleFileTransferResult> singleResults)
+    {
+        var success = singleResults.All(x => x.Success);
+        var actionSkipped = success && singleResults.All(x => x.ActionSkipped);
+        var userResultMessage = GetUserResultMessage(singleResults.ToList());
+
+        _logger.LogBatchFinished(_batchContext, userResultMessage, success, actionSkipped);
+
+        var transferErrors = singleResults.Where(r => !r.Success).GroupBy(r => r.TransferredFile ?? "--unknown--")
+                .ToDictionary(rg => rg.Key, rg => (IList<string>)rg.SelectMany(r => r.ErrorMessages).ToList());
+
+        var transferredFileResults = singleResults.Where(r => r.Success && !r.ActionSkipped).ToList();
+
+        var fileNames = from result in singleResults
+                        where result.Success
+                        select result.TransferredFile;
+
+        var filePaths = from result in singleResults
+                        where result.Success
+                        select result.TransferredFilePath;
+
+        return new FileTransferResult {
+            ActionSkipped = actionSkipped,
+            Success = success,
+            UserResultMessage = userResultMessage,
+            SuccessfulTransferCount = singleResults.Count(s => s.Success && !s.ActionSkipped),
+            FailedTransferCount = singleResults.Count(s => !s.Success && !s.ActionSkipped),
+            TransferredFileNames = transferredFileResults.Select(r => r.TransferredFile ?? "--unknown--").ToList(),
+            TransferErrors = transferErrors,
+            TransferredFilePaths = transferredFileResults.Select(r => r.TransferredFilePath ?? "--unknown--").ToList(),
+            OperationsLog = new Dictionary<string, string>() 
+        };
+    }
 
-            return userResultMessage;
-        }
+    /// <summary>
+    /// Forms the userResultMessage for FileTransferResult object
+    /// </summary>
+    /// <param name="results"></param>
+    /// <returns></returns>
+    private string GetUserResultMessage(IList<SingleFileTransferResult> results)
```

Comment:
This method can be static, since it doesn't access instance data.

Reply:
Fixed